### PR TITLE
chore: updates translation files and better transifex mapping

### PR DESCRIPTION
**Pull request by eLRuLL**
improves translation for nl, ja, de, ko, it, pl, tr, es_419, hi and zh_CN languages

**Testing instructions:**
- Create a new course and add a new DnDv2 problem
- Edit the DnDv2 problem to use the Assessment mode
- Change the user's language going to /update_lang/ (lms), choose one supported by this xblock (check "LANGUAGES" inside this repository settings file)
- Go back to the problem and confirm the problem is being translated correctly (`nl` and `ja` were not supported before, so they should be available now for example)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @eLRuLL! I've created [OSPR-5690](https://openedx.atlassian.net/browse/OSPR-5690) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@eLRuLL Thank you for your contribution. Please let me know once it is ready for our review.

File: drag_and_drop_v2/public/js/translations/eo/text.js
Comment (pomegranited) [inline drag_and_drop_v2/public/js/translations/eo/text.js:19]: @eLRuLL  This shoudn't be removed. The `eo` language code here is used as a ["dummy" translation](https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/271/files#diff-d0a28b90104be3005d4da328f9991ce959800fcedb329cc503f89057e181d15cR88), for testing.

`eo` is what [edx uses](https://github.com/edx/edx-platform/blob/b4614d75f0eb56861388ca31ae8cfa840a0c9374/conf/locale/config.yaml#L95-L98) too, but it's unfortunate, since `eo` is `Esperanto`.

You could change the `drag_and_drop_v2/translations/config.yaml` file to use the word "dummy" like we do in LX.

But either way, you need to set up a virtualenv and run these commands to reinstate the dummy translation files.

```
pip install -r requirements.txt
make build_dummy_translations
```

**Review by pomegranited — CHANGES_REQUESTED** [review]

File: drag_and_drop_v2/translations/ar/LC_MESSAGES/text.po
Comment (pomegranited) [inline drag_and_drop_v2/translations/ar/LC_MESSAGES/text.po:274]: It's hard to tell, but it doesn't look like the Arabic translations here don't match the ones in the spreadsheet?

* [Some of your answers were not correct](https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/271/files#diff-fe2f51c0844ee88f72aac3c278df8aa0754169a21e14ea6b429c47e5eead767cR597)
* [Problem)(https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/271/files?file-filters%5B%5D=.js&file-filters%5B%5D=.md&file-filters%5B%5D=.mo&file-filters%5B%5D=.po&file-filters%5B%5D=.py&file-filters%5B%5D=No+extension#diff-fe2f51c0844ee88f72aac3c278df8aa0754169a21e14ea6b429c47e5eead767cR669)
* [Correctly placed {correct_count} items](https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/271/files?file-filters%5B%5D=.js&file-filters%5B%5D=.md&file-filters%5B%5D=.mo&file-filters%5B%5D=.po&file-filters%5B%5D=.py&file-filters%5B%5D=No+extension#diff-fe2f51c0844ee88f72aac3c278df8aa0754169a21e14ea6b429c47e5eead767cR434)
* [Misplaced {misplaced_count} items](https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/271/files?file-filters%5B%5D=.js&file-filters%5B%5D=.md&file-filters%5B%5D=.mo&file-filters%5B%5D=.po&file-filters%5B%5D=.py&file-filters%5B%5D=No+extension#diff-fe2f51c0844ee88f72aac3c278df8aa0754169a21e14ea6b429c47e5eead767cR444)

**Review by pomegranited — CHANGES_REQUESTED** [review]
@eLRuLL 

* The translations here don't all seem to match the ones on the spreadsheet for LX-1802. I marked the Arabic ones that looked incorrect, but could you please check the others too?
* Dutch and Japanese translations have been removed for some reason?

File: drag_and_drop_v2/translations/ar/LC_MESSAGES/text.po
Comment (eLRuLL) [reply drag_and_drop_v2/translations/ar/LC_MESSAGES/text.po:274]: @pomegranited I didn't touch the translations that already appeared as reviewed in edX transifex, should I change them to the ones recommended by LX?

Comment (pomegranited) [reply drag_and_drop_v2/translations/ar/LC_MESSAGES/text.po:274]: Ah, so you're pulling from Transifex, that's good. You or LX should update the translations in Transifex to match the spreadsheet, and then you can pull them again. Will comment on that ticket so you know who to talk to there.

You might need to help with the ones that have variables in them, since the spreadsheet I can see doesn't use the right variable names or syntax.

Comment (eLRuLL) [reply drag_and_drop_v2/translations/ar/LC_MESSAGES/text.po:274]: @pomegranited yeah I was trying to say that I intentionally didn't change those because I thought it was a better idea to stay with the ones added by edX, but I can make the changes so I'll update them now

Comment (eLRuLL) [reply drag_and_drop_v2/translations/ar/LC_MESSAGES/text.po:275]: updated now :+1:

**Comment by pomegranited** [comment]
@eLRuLL  This [labxchange-dev#1353](https://gitlab.com/opencraft/client/LabXchange/labxchange-dev/-/merge_requests/1353) + [edx-platform#27162](https://github.com/edx/edx-platform/pull/27162)  is working in LX for the following languages:
* Arabic (ar-ae)
* Chinese (zh-hans)
* French (fr)
* Italian (it) -- working but missing a couple strings:
   italian_missing
* Korean (ko)
* Portuguese (pt-br)
* Spanish (es-419)
* Turkish (tr-tr) -- working but missing a couple strings:
  turkish_missing

But not for these (English shows instead, except for German, which shows an error):
* Dutch (nl)
* German (de) -- actually shows an error in LX? "Es ist ein Fehler aufgetreten. Drag&Drop-Fragestellung kann nicht geladen werden."
* Japanese (ja)


Can you check these and figure out why?

**Comment by eLRuLL** [comment]
@pomegranited ty so much for the review, it has been super helpful:

- it: LX didn't provide translations for the missing strings
- tr: same, LX didn't provide those
- `nl` and `ja` were missing I think because they only have a few translations available, but I improved on those.
- `de`: one of the translations had an incorrect string match translated, it's fixed now

**Comment by pomegranited** [comment]
:+1: Looks great now, thank you @eLRuLL !

We'll likely need to do this again when LX updates the translations (Dutch and Italian specifically is missing a lot), but this is a good start.

- [x] I tested this on my edX and LX devstacks.
   Both devstacks:
   * Ensure that the following languages are enabled in the Django Admin DarkLangConfig as either allowed languages or beta languages: `en,ar,zh,nl,fr,de,it,ja,ko,pt,es,tr`.
   On Open edX devstack: 
   * Create a course, and publish a DnDv2 unit using Assessment mode and the default/demo activity.
   * Log in as a student and enroll in the course
   * View the DnDv2 module, and answer some elements correctly, some incorrectly.
   * For the configured language regions:
      (Note that because [Open edX supports only some regions for these languages](https://github.com/edx/edx-platform/blob/open-release/koa.2a/lms/envs/common.py#L1442-L1524), I had to use these specific language codes:  `ar,zh-cn,nl-nl,fr,de-de,it-it,ja-jp,ko-kr,pt-br,es-419,tr-tr`.)
      * Change the student's language using the darklang endpoint: http://edx.devstack:18000/update_lang/
      * Refresh the DnDv2 module
      * Ensure that the expected translated strings show up in the assessment bar, with expected formatting.
   On LabXchange devstack:
   * Login as a learner
   * Visit the DnDv2 activity (http://labxchange.devstack:4556/library/items/lb:LabXchange:e9531f21:drag-and-drop-v2:6-3)
   * View the DnDv2 module, and answer some elements correctly, some incorrectly.
   * For each of the available languages (`ar-ae,zh-hans,nl,fr,de,it,ja,ko,pt-br,es-419,tr-tr`):
      * Change the user's preferred language from the settings page: http://labxchange.devstack:4556/account-settings
      * Refresh the DnDv2 activity
      * Ensure that the expected translated strings show up in the assessment bar, with expected formatting.
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A
- [x] <strike>I made sure any change in configuration variables is reflected in the corresponding
    client's `configuration-secure` repository.</strike> N/A

**Review by pomegranited — APPROVED** [review]

**Comment by eLRuLL** [comment]
ty for the review @pomegranited 

@natabene this is ready, @hasnain-naveed could you please review this? It adds translation improvements, also updated the changelog even including previous releases, and bumped the version. I'll have to create another PR against edx-platform to increase the version there as well after this is merged and the new release created

**Comment by natabene** [comment]
@shafqatfarhan Could you please have a look? Please hold off with merging, I will let know when it is good to merge.

**Comment by eLRuLL** [comment]
@natabene I think this is good to merge, or is there something else needed?

**Comment by natabene** [comment]
@eLRuLL Yes, but we need to wait on one minor thing from Legal due to a contract end. Once that is sorted out, I will post here and anyone can merge.

**Comment by pomegranited** [comment]
Hi @natabene , how goes the contract stuff?

**Comment by natabene** [comment]
@pomegranited Still in the works, hope to post update soon.

**Comment by pomegranited** [comment]
Hi @natabene , any updates here?

**Comment by natabene** [comment]
@pomegranited We are good now. @shafqatfarhan Can you please merge when you have a chance?

**Comment by natabene** [comment]
@pomegranited Feel free to merge, if you can.

**Comment by pomegranited** [comment]
Thank you @natabene ! I can, and did :)

FYI @eLRuLL

**Comment by edx-webhook** [comment]
@eLRuLL 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx-solutions/xblock-drag-and-drop-v2&entry.867055334=https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/271&entry.1484655318=https://github.com/eLRuLL&entry.752974735=2021-03-25+22:56&entry.1917517419=2021-05-25+02:49&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
